src/library/tracksuggestion/tracksuggestionfeature.cpp

```diff
@@ -0,0 +1,416 @@
+#include "library/tracksuggestion/tracksuggestionfeature.h"
+
+#include "library/dao/trackschema.h"
+#include "library/library.h"
+#include "library/queryutil.h"
+#include "library/trackcollectionmanager.h"
+#include "library/treeitem.h"
+#include "mixer/playerinfo.h"
+#include "mixer/playermanager.h"
+#include "track/track.h"
+#include "widget/wlibrary.h"
+#include "widget/wlibrarytextbrowser.h"
+
+namespace {
+
+const QString composeTreeItemLabel(const QString& track,
+        const QString& artist) {
+    return track + QStringLiteral(" | ") + artist;
+}
+
+const QString kSuggestionTitle("SuggestionView");
+
+// This can be replaced with the XML parse.
+const QString kTrack = "track";
+const QString kName = "name";
+const QString kPlaycount = "playcount";
+const QString kMbid = "mbid";
+const QString kMatch = "match";
+const QString kDuration = "duration";
+const QString kArtist = "artist";
+const QString kUrl = "url";
+const QString kImage = "image";
+
+// Dummy location added: If there is no location for a suggestion, seg fault occurs after pressing on the suggestion.
+const QString kLocation = "/localhost/Users/A/Music/Suggestion/Suggestion-Library/Tracks/";
```

Comment:
Ah, so this is only a workaround for a segfault. So we should probably track that down instead.